packages/kbn-shared-ux-components/src/page_template/with_solution_nav.tsx

```diff
@@ -38,11 +38,13 @@ export const withSolutionNav = (WrappedComponent: ComponentType<KibanaPageTempla
       // Have to store it as the opposite of the default we want
       localStorage.setItem(SOLUTION_NAV_COLLAPSED_KEY, JSON.stringify(isSideNavOpenOnDesktop));
     };
+
+    const { canBeCollapsed = true } = solutionNav;
     const sideBarClasses = classNames(
       'kbnPageTemplate__pageSideBar',
       {
         'kbnPageTemplate__pageSideBar--shrink':
-          isMediumBreakpoint || (isLargerBreakpoint && !isSideNavOpenOnDesktop),
+          isMediumBreakpoint || (isLargerBreakpoint && !isSideNavOpenOnDesktop && canBeCollapsed),
```

Comment:
Nit: I'd put the new variable as the first check since it supersedes the others
```suggestion
          isMediumBreakpoint || (canBeCollapsed && isLargerBreakpoint && !isSideNavOpenOnDesktop),
```